app/policies/activity_policy.rb

```diff
@@ -1,30 +1,46 @@
 class ActivityPolicy < ApplicationPolicy
   def index?
-    user.administrator? || user.fund_manager?
+    true
   end
 
   def show?
-    user.administrator? || user.fund_manager?
+    user_can_access?
   end
 
   def create?
-    user.administrator? || user.fund_manager?
+    user_can_create_and_update?
   end
 
   def update?
-    user.administrator? || user.fund_manager?
+    user_can_create_and_update?
   end
 
   def destroy?
     user.administrator? || user.fund_manager?
   end
 
+  private def associated_user?
+    user.organisation.id.eql?(record.organisation_id)
+  end
+
+  private def user_can_access?
+    return true if user.administrator? || user.fund_manager?
+    return true if record.class.name.eql?("Class")
```

Comment:
This is a bit horrible. Is there a better way of telling if a record has actually been created? `record.persisted?` doesn't work here

Reply:
Perhaps `exists?`?

Reply:
From reading the commit and tests around this private method I'm unsure when or how we get into the pickle where a record we're trying to view doesn't exist. What scenario prompted this guard clause?

Reply:
You know what I can't even remember! I will remove it.